chain/network/src/peer/peer_actor.rs

```diff
@@ -433,19 +430,19 @@ impl PeerActor {
             }
         };
 
-        self.network_state
-            .view_client_addr
-            .send(view_client_message)
-            .into_actor(self)
-            .then(move |res, act, _ctx| {
+        ctx.spawn(wrap_future(self.network_state.view_client_addr.send(view_client_message)).then(
+            move |res, act: &mut PeerActor, _ctx| {
                 // Ban peer if client thinks received data is bad.
                 match res {
                     Ok(NetworkViewClientResponses::TxStatus(tx_result)) => {
-                        let body = Box::new(RoutedMessageBody::TxStatusResponse(*tx_result));
-                        let _ = act
-                            .network_state
-                            .peer_manager_addr
-                            .do_send(PeerToManagerMsg::RouteBack(body, msg_hash.unwrap()));
+                        let msg = act.network_state.sign_message(
+                            &act.clock,
+                            RawRoutedMessage {
+                                target: AccountOrPeerIdOrHash::Hash(msg_hash.unwrap()),
```

Comment:
Agreed. Added a TODO. You can send #7814 to me for review whenever it is ready, or I can create my own. It was on my todo list anyway.

Reply:
https://github.com/near/nearcore/pull/7814 I think is ready, probably best to land it after this PR to avoid conflits though, but really no strong opinion, whatever, as long as it gets in the master someday